File: src/main/java/com/lyaiya/softcode/mixin/thaumicjei/MixinThaumcraftJEIPlugin.java
package com.lyaiya.softcode.mixin.thaumicjei;

import com.buuz135.thaumicjei.ThaumcraftJEIPlugin;
import com.lyaiya.softcode.constant.KeyConstant;
import com.lyaiya.softcode.constant.ModIdConstant;
import com.lyaiya.softcode.util.TranslationKeyKt;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.ModifyArg;

@Mixin(
        value = ThaumcraftJEIPlugin.class,
        remap = false
)
abstract class MixinThaumcraftJEIPlugin {
    @ModifyArg(
            method = "register",
            at = @At(
                    value = "INVOKE",
                    target = "Lmezz/jei/api/IModRegistry;addIngredientInfo(Ljava/util/List;Lmezz/jei/api/recipe/IIngredientType;[Ljava/lang/String;"
                            + ")V",
                    ordinal = 0
            ),
            index = 2
    )
    private String[] modifyArgIngredientInfo0(String[] descriptionKeys) {
        String key = TranslationKeyKt.create(ModIdConstant.THAUMIC_JEI, KeyConstant.DESCRIPTION, "salis_mundis");
        return new String[]{key};
    }

    @ModifyArg(
            method = "register",
            at = @At(
                    value = "INVOKE",
                    target = "Lmezz/jei/api/IModRegistry;addIngredientInfo(Ljava/lang/Object;Ljava/lang/Class;[Ljava/lang/String;)V",
                    ordinal = 0
            ),
            index = 2
    )
    private String[] modifyArgIngredientInfo1(String[] descriptionKeys) {
        String key = TranslationKeyKt.create(ModIdConstant.THAUMIC_JEI, KeyConstant.DESCRIPTION, "triple_meat_treat");
        return new String[]{key};
    }
}

File: src/main/java/com/lyaiya/softcode/mixin/thaumicjei/MixinThaumcraftJEIPlugin.kt
package com.lyaiya.softcode.mixin.thaumicjei

import com.buuz135.thaumicjei.ThaumcraftJEIPlugin
import com.lyaiya.softcode.constant.KeyConstant
import com.lyaiya.softcode.constant.ModIdConstant
import com.lyaiya.softcode.util.createTranslationKey
import net.minecraft.client.resources.I18n
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.injection.Constant
import org.spongepowered.asm.mixin.injection.ModifyConstant

@Mixin(ThaumcraftJEIPlugin::class, remap = false)
abstract class MixinThaumcraftJEIPlugin {
    @ModifyConstant(
        method = ["register"],
        constant = [Constant(stringValue = "To create Salis Mundis, take 3 Vis Crystals of different types and combine them with Redstone Dust by crafting them with a flint and a bowl.")]
    )
    private fun modifyConstantRegister0(text: String): String {
        return I18n.format(createTranslationKey(ModIdConstant.THAUMIC_JEI, KeyConstant.DESCRIPTION, "salis_mundis"))
    }

    @ModifyConstant(
        method = ["register"],
        constant = [Constant(stringValue = "To create the Triple Meat Treat, take 3 different kinds of meat nuggets (produced by cooking meat in the Infernal Furnace) and mix them with sugar.")]
    )
    private fun modifyConstantRegister1(text: String): String {
        return I18n.format(createTranslationKey(ModIdConstant.THAUMIC_JEI, KeyConstant.DESCRIPTION, "triple_meat_treat"))
    }
}